extra_data/tests/test_reader_mockdata.py

```diff
@@ -743,15 +743,49 @@ def test_train_timestamps(mock_scs_run):
     assert np.all(np.diff(tss).astype(np.uint64) > 0)
 
     # Convert numpy datetime64[ns] to Python datetime (dropping some precision)
-    dt0 = tss[0].astype('datetime64[ms]').item().replace(tzinfo=timezone.utc)
+    tss_l = run.train_timestamps(pydatetime=True)
+    assert len(tss_l) == len(run.train_ids)
     now = datetime.now(timezone.utc)
-    assert dt0 > (now - timedelta(days=1))  # assuming tests take < 1 day to run
-    assert dt0 < now
+    assert tss_l[0] > (now - timedelta(days=1))  # assuming tests take < 1 day to run
+    assert tss_l[0] < now
+    assert tss_l[0].tzinfo is timezone.utc
 
     tss_ser = run.train_timestamps(labelled=True)
     assert isinstance(tss_ser, pd.Series)
     np.testing.assert_array_equal(tss_ser.values, tss)
     np.testing.assert_array_equal(tss_ser.index, run.train_ids)
+    assert tss_ser.dt.tz is timezone.utc
+
+def test_train_timestamps_local_time(mock_scs_run):
+    run = RunDirectory(mock_scs_run)
+
+    del1h = timedelta(hours=1)
+    del2h = timedelta(hours=2)
+      
+    # First, the pydatetime case
+    tss = run.train_timestamps(pydatetime=True)
+    tss_berlin = run.train_timestamps(pydatetime=True, euxfel_local_time=True)
+
+    # The time difference between UTC and Europe/Berlin can only be
+    # one or two hours depending on daylight savings
+    assert all(
+        t2.replace(tzinfo=None) - t1.replace(tzinfo=None) == del1h or
+        t2.replace(tzinfo=None) - t1.replace(tzinfo=None) == del2h
```

Comment:
The `datetime` object has a `.utcoffset()` method you can use instead of these subtractions - it also gives a timedelta.

Reply:
Good point, the case with `euxfel_local_time=False` is tested separately so it's safe to assume that `t1` in this case is always correct and therefore sufficient to check the UTC offset. Will consolidate all your suggestions into a commit and fix this at the same time. Thanks!